sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -207,62 +205,59 @@ internal string QueryAnswerRaw
 
                 if (QueryAnswer.HasValue)
                 {
-                    queryAnswerStringValue = $"{QueryAnswer.Value}{QueryAnswerCountRawSplitter}{QueryAnswerCount.GetValueOrDefault(1)}{QueryAnswerThresholdRawSplitter}{QueryAnswerThreshold.GetValueOrDefault(0.7)}";
+                    var queryAnswerValues = new List<string>();
+
+                    if (QueryAnswerCount.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerCountRaw}{QueryAnswerCount}");
+                    }
+
+                    if (QueryAnswerThreshold.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerThresholdRaw}{QueryAnswerThreshold}");
+                    }
+
+                    queryAnswerStringValue = $"{QueryAnswer.Value}" + (queryAnswerValues.Count > 0 ? $"|{string.Join(",", queryAnswerValues)}" : "");
                 }
 
                 return queryAnswerStringValue;
             }
-
             set
             {
-                if (string.IsNullOrEmpty(value))
-                {
-                    QueryAnswer = null;
-                    QueryAnswerCount = null;
-                    QueryAnswerThreshold = null;
-                }
-                else
+                if (!string.IsNullOrEmpty(value)) // If the value is - "extractive" or "extractive|count-1" or "extractive|threshold-0.7" or "extractive|count-5,threshold-0.9" or "extractive|threshold-0.8,count-4"
                 {
-                    if (value.Contains(QueryAnswerCountRawSplitter) || value.Contains(QueryAnswerThresholdRawSplitter))
+                    string[] queryAnswerValues = value.Split('|');
```

Comment:
can there be spaces or others? If so, should we guard against it?

Reply:
There will be not spaces. The format is "extractive|count-5,threshold-0.7"